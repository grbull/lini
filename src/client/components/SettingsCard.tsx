import cn from 'classnames';
import React, { ReactElement, ReactNode } from 'react';

interface Props {
  children: ReactNode;
  title: string;
}

export function SettingsCard({ children, title }: Props): ReactElement {
  return (
    <div className={cn('p-2.5', 'py-3')}>
      <h2 className={cn('pb-2', 'text-lg', 'font-bold')}>{title}</h2>
      <div className={cn()}>{children}</div>
    </div>
  );
}
