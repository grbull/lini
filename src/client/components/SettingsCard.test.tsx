/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom/extend-expect';

import { render, RenderResult } from '@testing-library/react';
import React from 'react';

import { SettingsCard } from './SettingsCard';

function setup(): RenderResult {
  const utils = render(
    <SettingsCard title="Testing">More testing</SettingsCard>
  );

  return { ...utils };
}

describe('SettingsCard Component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = setup();
    expect(asFragment()).toMatchSnapshot();
  });
});
