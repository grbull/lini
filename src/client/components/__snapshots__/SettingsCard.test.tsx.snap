// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsCard Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="p-2.5 py-3"
  >
    <h2
      class="pb-2 text-lg font-bold"
    >
      Testing
    </h2>
    <div
      class=""
    >
      More testing
    </div>
  </div>
</DocumentFragment>
`;
